### PR TITLE
feat: add rollup config

### PR DESCRIPTION
**Pull request by domhorvathtw**
Part of #23 umbrella PR. 

## ❓ Context
To create a **UMD** and **ES** bundle this PR adds rollup to the build process and builds bundles from the generated react icon components into the `lib` folder.

## 🚀 Changes
- rollup config
- add npm scripts
- eslintignore to ignore build files (generated react icon components)

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      },
+    },
+    {
+      file: pkg.module,
+      format: 'es',
+    },
+  ],
+  external: [
+    ...Object.keys(pkg.dependencies || {}),
+    ...Object.keys(pkg.devDependencies || {}),
```
Comment (pterblgh) [inline rollup.config.js]: Why do you need `devDependencies` in externals? 🤔 

In my understanding `devDependencies` should only contain those packages whose are must have to be able to build/develop your library - like `rollup`.

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    ...Object.keys(pkg.peerDependencies || {}),
+  ],
+  plugins: [
+    typescript({
+      // eslint-disable-next-line global-require
+      typescript: require('typescript'),
```
Comment (pterblgh) [inline rollup.config.js]: Why do you need to require the whole `typescript` package here? I briefly checked this package's options but I still don't understand what's the purpose of this import.

File: rollup.config.js (hunk 1 above)
Comment (domhorvathtw) [reply rollup.config.js]: I'ts not needed, you are right, I'll remove it

File: rollup.config.js (hunk 2 above)
Comment (domhorvathtw) [reply rollup.config.js]: When I was reading about rollup, found this blog post, and followed its advice to make sure rollup uses the locally installed typescript version.

> The plugins section is a bit weird. What we’re doing there is making rollup-plugin-typescript2 use the locally-installed TypeScript. By default, it uses a version that likely isn’t up-to-date ...

https://hackernoon.com/building-and-publishing-a-module-with-typescript-and-rollup-js-faa778c85396

File: .circleci/config.yml [hunk 3]
```diff
@@ -56,50 +59,24 @@ jobs:
...
-      - run:
-          name: Deploy demo
-          command: |
-            mkdir ~/.ssh
-            ssh-keyscan github.com >> ~/.ssh/known_hosts
-            git config --global user.email "circleci@transferwise.com"
-            git config --global user.name "CircleCI"
-            npm run deploy-demo
+          name: Publish on npm
+          command: npm publish --access=public
```
Comment (pterblgh) [inline .circleci/config.yml]: So we're making this icon set publicly available to anyone to be able to consume from NPM?

Comment (domhorvathtw) [reply .circleci/config.yml]: It was public before, don't see any reason why should we should we change that. 
Probably need to update the licence field in package.json.
@oliverviljamaa what do you think?

Comment (oliverviljamaa) [reply .circleci/config.yml]: I see no reason to make it private either — the repo has been public from the start and neptune-web is public as well 👍

File: .circleci/config.yml [hunk 4]
```diff
@@ -108,37 +85,31 @@ workflows:
...
+      - build:
           requires:
             - test
-            - build-package
-          filters:
-            branches:
-              only: master
+            - dependencies
+      - publish-prerelease:
+        context: frontend-publish
```
Comment (oliverviljamaa) [inline .circleci/config.yml]: The indentation seems off.

File: package.json [hunk 5]
```diff
@@ -2,17 +2,25 @@
...
   "scripts": {
     "build": "run-s build:*",
     "build:clean": "rimraf lib build && mkdir lib && mkdir -p build/components",
     "build:react-components": "yarn run build-react-components",
     "build:format": "yarn prettier --write",
-    "test": "yarn run lint",
+    "build:rollup": "rollup -c",
+    "rollup:watch": "rollup -cw",
+    "test": "run-s test:*",
+    "test:lint": "yarn run lint",
```
Comment (oliverviljamaa) [inline package.json]:
```suggestion
    "test:lint": "yarn lint",
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -46,8 +48,9 @@ jobs:
           command: |
             mkdir ~/.ssh
             ssh-keyscan github.com >> ~/.ssh/known_hosts
-            npm run release-to-github
-  publish:
+            yarn run release-to-github
```
Comment (oliverviljamaa) [inline .circleci/config.yml]:
```suggestion
            yarn release-to-github
```

File: .circleci/config.yml (hunk 4 above)
Comment (domhorvathtw) [reply .circleci/config.yml]: Fixed 👍

**Pull request merged**